db/migrate/20210121183741_backfill_new_application_reference_datetime_columns.rb

```diff
@@ -0,0 +1,35 @@
+class BackfillNewApplicationReferenceDatetimeColumns < ActiveRecord::Migration[6.0]
+  def up
+    ApplicationReference
+    .all
+    .each do |reference|
+      feedback_provided_at = reference.audits.where("audited_changes#>>'{feedback_status, 1}' = 'feedback_provided'").last&.created_at
+      feedback_refused_at = reference.audits.where("audited_changes#>>'{feedback_status, 1}' = 'feedback_refused'").last&.created_at
+      cancelled_at = reference.audits.where("audited_changes#>>'{feedback_status, 1}' = 'cancelled'").last&.created_at
+      cancelled_at_eoc_at = reference.audits.where("audited_changes#>>'{feedback_status, 1}' = 'cancelled_at_end_of_cycle'").last&.created_at
+      email_bounced_at = reference.audits.where("audited_changes#>>'{feedback_status, 1}' = 'email_bounced'").last&.created_at
+
+      reference.update!(
+        feeback_provided_at: feedback_provided_at,
```

Comment:
Maybe an `audit_comment` here would be good for posterity.

Reply:
💯  good shout